### ISSUE TITLE
Is this project active?

### ISSUE THREAD

**Issue by SolidSoftwareServices** [OrleansContrib/Orleankka#171]
Hi,
Quick question, Is this project being actively maintained?. If so, is there a roadmap I can look at?

Thank you

Regards

**Comment by yevhen** [comment]
Hi! The project is not actively developed at the moment. The upstream project has made an attempt to subsume some of the functionality. The 4.x milestone of Orleans should bring some of the features like ease unit testing and POCO actors. 

I'm considering lifting FSM support into its own project but that is all at the moment. What kind of functionality you have in mind when asking for the roadmap?

**Comment by NTTAKR** [comment]
I just read about Orleankka and watched the video. I have started to implement something similar in Orleans like you have done with a single Interface being implemented in all grains and having a sort of command or message delivered with derived classes. It's not at all so sophisticated like what you have done, but I am missing some things in Orleans that I do find you have implemented in Orleankka. So it would be a shame if this project would not evolve further. 

In my mind a mixture between Akka and Orleans would be the best infrastructure to use. 
E.g. the Actor hierarchy is something I am missing in Orleans. Having a "responsible" Actor that get's a message if a child actor throws is a better concept than having the exception delivered to whomever called the grain. 

FSM support in Akka was also a thing I missed in Orleans, but I have yet to look at Orleankka and its implementation of that. 

How stable would you rate the current state of Orleankka? I try to decide if I would dare to use it in a production environment or if I rather should use Orleans. 

Thanks
  Andreas

**Comment by yevhen** [comment]
@NTTAKR I use Orleankka in production for 5 years already. No issues so far.  Yes, it's still active. Just recently I've added support for calling actors over HTTP (ASP.NET Core for server-side and custom HttpActorSystem) and planning to enhance it further with functionality that should allow writing durable workflows in a more straightforward way (ala Temporal.IO).

Wrt to FSM: I've implemented support for HSM in Orleankka which is even more powerful than one in Akka.Net.

For supervision hierarchy - it only makes sense with fire-n-forget communication, IMHO. But if you really need it - we can decide on the API and add it to Orleankka.
